src/components/admin-management/admin-form/AdminForm.tsx

```diff
@@ -0,0 +1,330 @@
+    onSubmit,
+    setValue,
+    getValues,
+    trigger,
+    setError,
+    isEditingExistingAdmin,
+    touchedFields,
+    adminId: selectedAdmin?.id,
+  });
+
+  return (
+    <MainWrapper>
+      <StyledFormTitle>
+        <ArrowBackIcon onClick={handleArrowBackClick} cursor="pointer" />
+        {translate('adminManagement.title')}
+      </StyledFormTitle>
+      <ManagementWrapper>
+        <Stack direction="row" justifyContent="space-between" mb={3}>
+          <PageName>
+            <Cylinder />
+            {isEditingExistingAdmin
+              ? translate('adminManagement.adminForm.changeAdminTitle')
+              : translate('adminManagement.adminForm.createAdminTitle')}
+          </PageName>
+        </Stack>
+        <Stack>
+          <form onSubmit={handleSubmit(handleSubmitClick)}>
+            <Grid container spacing={2}>
+              <Grid item xs={6}>
+                <FormControl fullWidth variant="filled">
+                  <InputLabel htmlFor="firstName">
+                    {translate('adminManagement.adminForm.firstName')}
+                  </InputLabel>
+                  <FilledInput
+                    {...register('firstName')}
+                    id="firstName"
+                    error={!!errors.firstName}
+                  />
+                  {errors?.firstName && <ErrorMessage>{errors.firstName?.message}</ErrorMessage>}
+                </FormControl>
+              </Grid>
+
+              <Grid item xs={6}>
+                <FormControl fullWidth variant="filled">
+                  <InputLabel htmlFor="lastName">
+                    {translate('adminManagement.adminForm.lastName')}
+                  </InputLabel>
+                  <FilledInput {...register('lastName')} id="lastName" error={!!errors.lastName} />
+                  {errors?.lastName && <ErrorMessage>{errors.lastName?.message}</ErrorMessage>}
+                </FormControl>
+              </Grid>
+              <Grid item xs={6}>
+                <FormControl fullWidth variant="filled">
+                  <InputLabel htmlFor="email">
+                    {translate('adminManagement.adminForm.email')}
+                  </InputLabel>
+                  <FilledInput
+                    {...register('email')}
+                    id="email"
+                    error={!!errors.email}
+                    disabled={isEditingExistingAdmin}
+                  />
+                  {errors?.email && <ErrorMessage>{errors.email?.message}</ErrorMessage>}
+                </FormControl>
+              </Grid>
+              <Grid item xs={6}>
+                <FormControl disabled fullWidth variant="filled">
+                  <Controller
+                    name="phoneNumber"
+                    control={control}
+                    render={({ field, fieldState }): JSX.Element => (
+                      <MuiTelInput
+                        {...field}
+                        onChange={(value: string): void => {
```

Comment:
To be honest I'm not sure if this function should be inline, it's just too large to be directly in the component

Reply:
fixed